### PR TITLE
fix: Model should ignore answering if user ask queries like List all the documents

### PR DESCRIPTION
**Pull request by Harmanpreet-Microsoft**
## Purpose
 Model should ignore answering if user ask queries like mentioned below,
-List all the documents
-Give all the file names or documents name which are in the repository.

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code
Ask the model any of the following questions:
-List all the documents
-Give all the file names or documents name which are in the repository.
-List the Topic you can help me with

## What to Check
The model should ignore answering these questions and reply I cant list the documents 

## Other Information
image

### CODE REVIEW

File: code/create_app.py [hunk 1]
```diff
@@ -436,6 +436,7 @@ def speech_config():
             if response.status_code == 200:
                 return {
                     "token": response.text,
+                    "key": speech_key,
```
Comment (adamdougal) [inline code/create_app.py:439]: Is this required? We already have an endpoint in `get_speech_key` which returns the short lived key.

File: code/frontend/src/components/Answer/Answer.tsx [hunk 2]
```diff
@@ -24,23 +31,51 @@ export const Answer = ({
...
     const [chevronIsExpanded, setChevronIsExpanded] = useState(isRefAccordionOpen);
     const refContainer = useRef<HTMLDivElement>(null);
+    const [audioContext, setAudioContext] = useState<AudioContext | null>(null); //Manully  manage the audio context eg pausing resuming
+    // const [synthesizer, setSynthesizer] = useState<sdk.SpeechSynthesizer | null>(null);
     const handleChevronClick = () => {
         setChevronIsExpanded(!chevronIsExpanded);
         toggleIsRefAccordionOpen();
       };
 
+    const fetchSpeechConfig = async (): Promise<{ key: string; region: string }> => {
```
Comment (adamdougal) [inline code/frontend/src/components/Answer/Answer.tsx:49]: These changes look unrelated to the PR. Are they requried for something else?

File: code/backend/batch/utilities/helpers/config/default.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
...
...nse, embedding the extracted facts. Attribute the data to the corresponding document using the citation format [doc+index]. Strive to achieve a harmonious blend of brevity, clarity, and precision, maintaining the contextual relevance and consistency of the original source. Above all, confirm that your response satisfies the user's query with accuracy, coherence, and user-friendly composition.\n## Very Important Instruction\n- **You must generate the citation for all the document sources you have refered at the end of each corresponding sentence in your response.\n- If no documents are provided, **you cannot generate the response with citation**,\n- The citation must be in the format of [doc+index].\n- **The citation mark [doc+index] must put the end of the corresponding sentence which cited the document.**\n- **The citation mark [doc+index] must not be part of the response sentence.**\n- **You cannot list the citation at the end of response.\n- Every claim statement you generated must have at least one citation.**\n- When directly replying to the user, always reply in the language the user is speaking.\n- You **must not** respond if asked to List all documents in your repository.",
```
Comment (adamdougal) [inline code/backend/batch/utilities/helpers/config/default.json:5]: > You **must not** respond if asked to List all documents in your repository.

What does the LLM return when asked? It sounds like this will get it to return nothing. Would it instead be better return a message saying something like "I am unable to return the requested information. Please try another query or topic."

**Review by adamdougal — CHANGES_REQUESTED** [review]

File: code/create_app.py (hunk 1 above)
Comment (Harmanpreet-Microsoft) [reply code/create_app.py]: reverted the made changes

File: code/frontend/src/components/Answer/Answer.tsx (hunk 2 above)
Comment (Harmanpreet-Microsoft) [reply code/frontend/src/components/Answer/Answer.tsx]: reverted the made changes

File: code/backend/batch/utilities/helpers/config/default.json (hunk 3 above)
Comment (Harmanpreet-Microsoft) [reply code/backend/batch/utilities/helpers/config/default.json]: Whenever the user asks to " List all documents in your repository " the model reply as " I'm sorry, but I can't list all documents in the repository. If you have any specific questions or need information on a particular topic, feel free to ask!  "
image

File: code/backend/batch/utilities/orchestrator/open_ai_functions.py [hunk 4]
```diff
@@ -66,6 +66,7 @@ async def orchestrate(
         Call the text_processing function when the user request an operation on the current context, such as translate, summarize, or paraphrase. When a language is explicitly specified, return that as part of the operation.
         When directly replying to the user, always reply in the language the user is speaking.
         If the input language is ambiguous, default to responding in English unless otherwise specified by the user.
+        You **must not** respond if asked to List all documents in your repository.
```
Comment (Roopan-Microsoft) [inline code/backend/batch/utilities/orchestrator/open_ai_functions.py:69]: Please try to update a generic message and also try with all the possibilities of asking questions for the same scenario.

**Review by adamdougal — APPROVED** [review]
